From 7452d38da8a9b0cb826e7553b77da8dd56d2e330 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 11 Sep 2023 17:37:12 +0200
Subject: [PATCH 40/65] serial: stm32: adapt timeout to STM32MP2

STM32MP2 USART has a FIFO of 64 bytes. Timeouts should be longer to fit
with this FIFO size.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 06713f23ed85111a56854d5abbb8cebf17904362 serial: stm32: adapt timeout to STM32MP2]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: Idee5a6510674004d9b9576fc267b5b6f01e16f60
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/326978
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/serial/serial_stm32.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/serial/serial_stm32.c b/drivers/serial/serial_stm32.c
index 1d05790e9ae..a87414a43a1 100644
--- a/drivers/serial/serial_stm32.c
+++ b/drivers/serial/serial_stm32.c
@@ -28,7 +28,9 @@
  * 10 bits are needed for worst case (8 bits + 1 start + 1 stop) = 86.806 us
  */
 #define ONE_BYTE_B115200_US		87
-#define STM32_USART_FIFO_TMO_US		(16 * ONE_BYTE_B115200_US)
+
+/* This is used to compute a timeout, take the worst possible case: STM32MP2 */
+#define STM32_USART_FIFO_TMO_US		(64 * ONE_BYTE_B115200_US)
 
 static void _stm32_serial_setbrg(void __iomem *base,
 				 struct stm32_uart_info *uart_info,
-- 
2.51.0

